src/Deployer.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.17;
+
+import {AddressDriver} from "./AddressDriver.sol";
+import {Caller} from "./Caller.sol";
+import {DripsHub} from "./DripsHub.sol";
+import {Managed, ManagedProxy} from "./Managed.sol";
+import {NFTDriver} from "./NFTDriver.sol";
+import {ImmutableSplitsDriver} from "./ImmutableSplitsDriver.sol";
+
+contract Deployer {
+    // slither-disable-next-line immutable-states
+    address public creator;
+
+    DripsHub public dripsHub;
+    bytes public dripsHubArgs;
+    uint32 public dripsHubCycleSecs;
+    DripsHub public dripsHubLogic;
+    bytes public dripsHubLogicArgs;
+    address public dripsHubAdmin;
+
+    Caller public caller;
+    bytes public callerArgs;
+
+    AddressDriver public addressDriver;
+    bytes public addressDriverArgs;
+    AddressDriver public addressDriverLogic;
+    bytes public addressDriverLogicArgs;
+    address public addressDriverAdmin;
+    uint32 public addressDriverId;
+
+    NFTDriver public nftDriver;
+    bytes public nftDriverArgs;
+    NFTDriver public nftDriverLogic;
+    bytes public nftDriverLogicArgs;
+    address public nftDriverAdmin;
+    uint32 public nftDriverId;
+
+    ImmutableSplitsDriver public immutableSplitsDriver;
+    bytes public immutableSplitsDriverArgs;
+    ImmutableSplitsDriver public immutableSplitsDriverLogic;
+    bytes public immutableSplitsDriverLogicArgs;
+    address public immutableSplitsDriverAdmin;
+    uint32 public immutableSplitsDriverId;
+
+    constructor(
+        uint32 dripsHubCycleSecs_,
+        address dripsHubAdmin_,
+        address addressDriverAdmin_,
+        address nftDriverAdmin_,
+        address immutableSplitsDriverAdmin_
+    ) {
```

Comment:
Why three different admins? I think for now we could simplify it. 
We only need one.

Reply:
> We could design is already in a more create2 readiness.

But why? We don't need CREATE2, it doesn't provide us anything while making deployment addresses **more** volatile because dependent on the deployed bytecode. If deployment is callable only once, we know upfront what's the nonce and we don't care about the bytecode.

> deploy method which is only callable once

We can't do that, because it'd require Deployer to contain a copy of bytecode of all the deployed contracts. It'd more or less double the deployment cost. Most importantly it'd be tricky to fit so much data in a single contract bytecode, it'd need to be somehow split.

> Why three different admins?

I didn't want to drop the ability to make admins different, e.g. owned by different addresses or made non-upgradeable. On one hand you're right, we probably are going to use one address in practice, but on the other it doesn't take much to support varied ownership. WDYT?

Reply:
> We can't do that, because it'd require Deployer to contain a copy of bytecode of all the deployed contracts. It'd more or less double the deployment cost. Most importantly it'd be tricky to fit so much data in a single contract bytecode, it'd need to be somehow split.

Ah I see. I was not considering this point. Then it makes sense to trigger it in the constructor

Reply:
I think we can also keep the three admins instead of changing it again.